### PR TITLE
Rims dowork

### LINKED ISSUE #28: Implement doWork for RIMS
Once we've got doWork workin' for the other Equipment, RIMS should be a piece of cake...

Note that this is pretty much dependent on #8 and #9 wrapping up.

### PR DESCRIPTION
**Pull request by azyth**
resolves #28  with two rims versions, I assume fancy rims will be the end result code but no sure if we want to have a simpler version or both?

### CODE REVIEW

File: lib/Ohmbrewer_RIMS.cpp [hunk 1]
```diff
@@ -245,12 +246,58 @@ bool Ohmbrewer::RIMS::isOff() const {
...
+        // turn off therm
+        getTube()->setState(false);
+        getTube()->work(); //reset
+    }*/
+
+    //FANCY RIMS, turns the pump off to rest when tun temp is good and safety temp is good.
+    if (getState()) {
+        //IF RIMS ON
+        // make sure R. PUMP is ON if safety temp > tun temp +3(margin)
+        if (getSafetyTemp()->c() > (getTube()->getSensor()->getTemp()->c() + 3) ){
```
Comment (kyleoliveira) [inline lib/Ohmbrewer_RIMS.cpp]: Why not use `getTunSensor()` instead of `getTube()->getSensor()` ?

File: lib/Ohmbrewer_RIMS.cpp [hunk 2]
```diff
@@ -245,12 +246,58 @@ bool Ohmbrewer::RIMS::isOff() const {
...
+            if (!getRecirculator()->getState()){ getRecirculator()->setState(true); }// turn on pump
```
Comment (kyleoliveira) [inline lib/Ohmbrewer_RIMS.cpp]: Break this if-statement into multiple lines for readability.

File: lib/Ohmbrewer_RIMS.cpp [hunk 3]
```diff
@@ -245,12 +246,58 @@ bool Ohmbrewer::RIMS::isOff() const {
...
+            getRecirculator()->work();
+        }
+        //if tun temp is less than 2 degrees (margin) of target temp, then: RIMS
+        if (getTube()->getSensor()->getTemp()->c() < (getTube()->getTargetTemp()->c() - 2) ){
+            // safetySensor guard on Therm (if: safety sensor temp > safety setting, then: NO heat )
+            if ( getSafetyTemp()->c() > getSafetySensor()->getTemp()->c() ) {
+                if (!getTube()->getState()) { getTube()->setState(true); } // turn on therm
```
Comment (kyleoliveira) [inline lib/Ohmbrewer_RIMS.cpp]: Break this if-statement into multiple lines for readability.

**Comment by azyth** [comment]
NOTE: all equipment (electrical relay type) should have work() called on tehm after they have had their state turned off to make sure the pin outpins are turned off.

File: lib/Ohmbrewer_RIMS.cpp (hunk 1 above)
Comment (azyth) [inline lib/Ohmbrewer_RIMS.cpp]: because i forgot it existed

**Comment by kyleoliveira** [comment]
Looks good to go.

**Pull request merged**